src/components/store/my-state.ts

```diff
@@ -0,0 +1,152 @@
+import { StateCreator } from "zustand";
+import { NewIngredient, NewRecipe, Product } from "../types.ts";
+
+export interface MyIngredientsState {
+  ingredients: NewIngredient[];
+  addIngredient: (ingredient: NewIngredient) => void;
+  editIngredients: (ingredients: NewIngredient[]) => void;
+}
+
+export interface MyRecipesState {
+  recipes: NewRecipe[];
+  addRecipe: (newRecipe: NewRecipe) => void;
+}
+
+export interface MyTemporaryUIState {
+  amount: number;
+  editAmount: (amount: number) => void;
+  name: string | null;
+  editName: (name: string | null) => void;
+  selectedIngredient: NewIngredient | null;
+  selectedIngredients: NewIngredient[];
+  selectedProducts: Product[];
+  selectIngredient: (ingredient: NewIngredient | null) => void;
+  addToSelectedIngredients: (ingredient: NewIngredient) => void;
+  editSelectedIngredients: (ingredients: NewIngredient[] | undefined) => void;
+  addToSelectedProducts: (product: Product) => void;
+  editSelectedProducts: (products: Product[] | undefined) => void;
+  deleteSelectedIngredient: (ingredient: NewIngredient) => void;
+  deleteSelectedProduct: (product: Product) => void;
+}
+
+export interface MyRecipesInCartState {
+  recipesInCart: string[];
+  addRecipeToCart: (recipe: string) => void;
+  deleteRecipeFromCart: (recipe: string) => void;
+}
+
+export const createIngredientsSlice: StateCreator<
+  MyIngredientsState,
+  [],
+  [],
+  MyIngredientsState
+> = (set) => ({
+  ingredients: [],
+  addIngredient: (ingredient: NewIngredient) => {
+    set((state) => ({
+      ingredients: [...state.ingredients, ingredient],
+    }));
+  },
+  editIngredients: (ingredients: NewIngredient[]) => {
+    set({
+      ingredients: ingredients,
+    });
+  },
+});
+
+export const createRecipesSlice: StateCreator<
+  MyRecipesState,
+  [],
+  [],
+  MyRecipesState
+> = (set) => ({
+  recipes: [],
+  addRecipe: (newRecipe: NewRecipe) => {
+    set((state) => ({
+      recipes: [...state.recipes, newRecipe],
+    }));
+  },
+});
+
+export const createTemporaryUISlice: StateCreator<
```

Comment:
Neco me ted napadlo, jak se nam to trochu rozrostlo, mozna by stalo za to ten stav perfixovat. eg. ui.name

to znamena ze pouziti by bylo:

```js
const { ui: {name} } = useMyStore()
```

Asi by bylo trochu jednodussi o tom premyslet, co myslis?